File: same.py
from urllib2 import Request, urlopen
from urllib import urlencode, quote_plus
url = 'https://api.shop.com:8443/AffiliatePublisherNetwork/v1/categories'
queryParams = '?' + urlencode({ quote_plus('publisherID') : 'TEST' ,quote_plus('locale') : 'en_US'  })
headers = {  'apikey':'l7xxa85a2511a8454491ac39f7a02cab7eb8'  }
request = Request(url + queryParams
, headers=headers)
request.get_method = lambda: 'GET'
response_body = urlopen(request).read()
print response_body

File: same.rb
require 'rubygems'
require 'rest_client'
require 'cgi'
url = 'https://api.shop.com/AffiliatePublisherNetwork/v1/categories'
headers  = { 'apikey' => 'l7xxa85a2511a8454491ac39f7a02cab7eb8' , :params => { CGI::escape('publisherID') => 'TEST',CGI::escape('locale') => 'en_US' } }
response = RestClient::Request.execute :method => 'GET', :url => url , :headers => headers
puts response
